### ISSUE TITLE
User-Agent string in embedded TV browsers

### ISSUE THREAD

**Issue by chrisn** [w3c/media-and-entertainment#47]
Web browsers are increasingly seeking to improve user privacy by reducing fingerprinting service, limit use of third party tracking cookies, etc.

One change that is relevant to embedded browsers in TVs is **freezing of the User-Agent string**.

Apple [introduced this a while ago in Safari][1], and Mozilla [support freezing the User Agent string][2]. For Google Chrome, refer to the following Chrome Platform status items:

* [Reduce User Agent string information][3]
* [User-Agent Client Hints][4]

and this [Intent to Deprecate][5] thread, also this [related thread][6] regarding the MacOS version number.

Google have proposed [**User-Agent Client Hints**][7] as a new mechanism for obtaining browser, device, and platform version information. This API has shipped in Chrome, with a view to freezing the User-Agent string at some time in the future. 

From Google's perspective, this is part of a larger set of changes, known as the [Privacy Sandbox][8].

As yet, it is not clear whether User-Agent Client Hints has support from other browser vendors.

With UA-CH, HTTP requests by default would only include the browser brand name and major version number, while other information is available on an opt-in basis - either by [setting an `Accept-CH` HTTP response header][9], or [calling a JavaScript API][10].

As many embedded TV browsers are based on Chromium, we recommend that TV browser integrators and TV standards groups that place requirements on content of the User-Agent string review their use of the User-Agent string and give feedback on the proposed changes.

MEIG is open to discussion on how TV devices and content providers can adapt to these proposed changes or to help coordinate sending media industry feedback.

A few questions that come to mind are:

* Does the UA-CH proposal cover all the information content providers need from a TV device?
* Where TV device standards define additional information to be included in the User Agent string (e.g., compliance with a particular TV standard revision), how should this information be conveyed using UA-CH?
* UA-CH includes a flag to indicate whether the user agent prefers a "mobile" user experience. Should there be similar for TV?
* If UA-CH sees broad adoption across desktop and mobile browsers, do we expect TV devices to also adopt it?
* etc.

Feedback can also be sent directly to the [User-Agent Client Hints GitHub issue tracker][11].

[1]: https://webkit.org/blog/8042/release-notes-for-safari-technology-preview-46/
[2]: https://mozilla.github.io/standards-positions/#ua-client-hints
[3]: https://www.chromestatus.com/feature/5704553745874944
[4]: https://chromestatus.com/feature/5995832180473856
[5]: https://groups.google.com/a/chromium.org/g/blink-dev/c/-2JIRNMWJ7s/m/yHe4tQNLCgAJ
[6]: https://groups.google.com/a/chromium.org/g/blink-dev/c/hAI4QoX6rEo/m/qQNPThr0AAAJ
[7]: https://wicg.github.io/ua-client-hints/
[8]: https://www.chromium.org/Home/chromium-privacy/privacy-sandbox
[9]: https://wicg.github.io/ua-client-hints/#examples
[10]: https://wicg.github.io/ua-client-hints/#getHighEntropyValues
[11]: https://github.com/WICG/ua-client-hints/issues

**Comment by chrisn** [comment]
Update: A `Sec-CH-UA-Form-Factor` hint has just been added, which allows for values such as "TV", "Mobile", "Tablet", "VR", etc. See https://github.com/WICG/ua-client-hints/issues/333

**Comment by anawhj** [comment]
Good point! It's been applied to https://wicg.github.io/ua-client-hints/#sec-ch-ua-form-factor by [#338](https://github.com/WICG/ua-client-hints/pull/338) last month.
It might require long times to adapt it to device maker and contents providers, but it will show several usages in future.
